File: converter/java/src/main/java/com/mlt/vector/flat/BooleanFlatVector.java
package com.mlt.vector.flat;

import com.mlt.vector.BitVector;
import com.mlt.vector.Vector;

import java.nio.ByteBuffer;

public class BooleanFlatVector extends Vector<ByteBuffer, Boolean> {
    private BitVector dataVector;

    public BooleanFlatVector(String name, BitVector dataVector) {
        super(name, dataVector.getBuffer());
        this.dataVector = dataVector;
    }

    public BooleanFlatVector(String name, BitVector nullabilityBuffer, BitVector dataVector) {
        super(name, nullabilityBuffer, dataVector.getBuffer());
        this.dataVector = dataVector;
    }

    @Override
    protected Boolean getValueFromBuffer(int index) {
        return this.dataVector.get(index);
    }

    public int size() {
        return this.dataVector.size();
    }
}

File: ts/src/vector/flat/booleanFlatVector.ts
import type BitVector from "./bitVector";
import Vector from "../vector";
import { type SelectionVector } from "../filter/selectionVector";

export class BooleanFlatVector extends Vector<Uint8Array, boolean> {
    private readonly dataVector: BitVector;

    constructor(name: string, dataVector: BitVector, sizeOrNullabilityBuffer: number | BitVector) {
        super(name, dataVector.getBuffer(), sizeOrNullabilityBuffer);
        this.dataVector = dataVector;
    }

    protected getValueFromBuffer(index: number): boolean {
        return this.dataVector.get(index);
    }
    override greaterThanOrEqualTo(value: boolean): SelectionVector {
        throw new Error("Method not available for type boolean");
    }
    override smallerThanOrEqualTo(value: boolean): SelectionVector {
        throw new Error("Method not available for type boolean");
    }
    override greaterThanOrEqualToSelected(value: boolean, selectionVector: SelectionVector): void {
        throw new Error("Method not available for type boolean");
    }
    override smallerThanOrEqualToSelected(value: boolean, selectionVector: SelectionVector): void {
        throw new Error("Method not available for type boolean");
    }
}
